python/ray/util/concurrent/futures/ray_executor.py

```diff
@@ -0,0 +1,658 @@
+        super().__init__(**kwargs)
+        self.index: int = 0
+        return
+
+    def next(self) -> _PoolActor:
+        """
+        Get the next indexed member of the actor pool
+
+        Returns
+        -------
+        _PoolActor
+            The current priority actor in the pool
+        """
+        obj = self.pool[self.index]
+        self.index += 1
+        if self.index >= len(self.pool):
+            self.index = 0
+        return obj
+
+
+@PublicAPI(stability="alpha")
+class RayExecutor(Executor):
+
+    """RayExecutor is a drop-in replacement for ProcessPoolExecutor and
+    ThreadPoolExecutor from concurrent.futures but distributes and executes
+    the specified tasks over a pool of dedicated actors belonging to a Ray
+    cluster instead of multiple processes or threads, respectively.
+
+    Attributes
+    ----------
+    max_workers : int | None
+        The number of actors to spawn. If max_workers=None, the work is
+        distributed over a number of actors equal to the number of CPU
+        resources observed by the Ray instance.
+    shutdown_ray : bool | None
+        Destroy the Ray cluster when self.shutdown() is called. If this is
+        None, RayExecutor will default to shutting down the Ray instance only
+        if it was initialised during instantiation of the current RayExecutor,
+        otherwise it will not be shutdown.
+    initializer : Callable[..., Any] | None
+        A function that will be called remotely in the actor context before the
+        submitted task (for compatibility with
+        concurrent.futures.ThreadPoolExecutor).
+    initargs : tuple[Any, ...] | None
+        Arguments for initializer (for compatibility with
+        concurrent.futures.ThreadPoolExecutor).
+    max_tasks_per_child : int | None
+        The maximum number of tasks to be performed by an actor before it is
+        gracefully killed and replaced (for compatibility with
+        concurrent.futures.ProcessPoolExecutor).
+    mp_context : Any | None
+        This is only included for compatibility with
+        concurrent.futures.ProcessPoolExecutor but is unused.
+    actor_pool_type: str | None
+        The type of actor pool to use: either 'random' (which will assign tasks
+        to randomly selected actors), or 'roundrobin' (which will simply assign
+        tasks to actors sequentially.
+
+    All additional keyword arguments are passed to ray.init()
+    (see https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-init).
+
+    For example, this will connect to a cluster at the specified address:
+    .. testcode::
+
+        RayExecutor(address='192.168.0.123:25001')
+
+    Note: excluding an address will initialise a local Ray cluster.
+    """
+
+    #: Aggregated Futures from initiated tasks.
+    futures: list[Future[Any]]
+    #: A container of Actor objects over which the tasks will be distributed.
+    actor_pool: _AbstractActorPool
+
+    def __init__(
```

Comment:
Would be nice to improve the validation on the arguments wherever it makes sense. Something like:
```
def __init__(self, ...):
    # ...existing code...
    
    # Better validation
    if max_workers is not None and max_workers < 1:
        raise ValueError(f"max_workers must be >= 1, got {max_workers}")
    
    if max_tasks_per_child is not None and max_tasks_per_child < 1:
        raise ValueError(f"max_tasks_per_child must be >= 1, got {max_tasks_per_child}")
    
    # Validate initializer early
    if initializer is not None and not callable(initializer):
        raise TypeError("initializer must be callable")
        
    # ...existing code...
```